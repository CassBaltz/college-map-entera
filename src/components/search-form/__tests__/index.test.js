import { shallow } from "enzyme";
import SearchForm from "..";

describe("SearchForm", () => {
  it("should render correctly - not searching", () => {
    const handleSubmit = jest.fn();
    const searching = false;
    const searchForm = shallow(
      <SearchForm searching={searching} handleSubmit={handleSubmit} />,
    );
    const button = searchForm.find("button");
    expect(button.prop("disabled")).toBe(false);
    expect(searchForm.getElement()).toMatchSnapshot();
  });

  it("should render correctly - searching", () => {
    const handleSubmit = jest.fn();
    const searching = true;
    const searchForm = shallow(
      <SearchForm searching={searching} handleSubmit={handleSubmit} />,
    );
    const button = searchForm.find("button");
    expect(button.prop("disabled")).toBe(true);
    expect(searchForm.getElement()).toMatchSnapshot();
  });

  it("should call its handleSubmit callback when the form is submitted", () => {
    const handleSubmit = jest.fn();
    const searching = true;
    const searchForm = shallow(
      <SearchForm searching={searching} handleSubmit={handleSubmit} />,
    );
    const form = searchForm.find("form");
    form.simulate("submit");
    expect(handleSubmit).toHaveBeenCalled();
  });
});
