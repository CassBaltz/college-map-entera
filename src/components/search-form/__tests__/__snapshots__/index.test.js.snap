// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should render correctly - not searching 1`] = `
<form
  className="schoolSearchForm"
  name="school-search"
  onSubmit={[MockFunction]}
>
  <input
    name="school-name"
    placeholder="Enter college name here"
    type="text"
  />
  <button
    disabled={false}
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`SearchForm should render correctly - searching 1`] = `
<form
  className="schoolSearchForm"
  name="school-search"
  onSubmit={[MockFunction]}
>
  <input
    name="school-name"
    placeholder="Enter college name here"
    type="text"
  />
  <button
    disabled={true}
    type="submit"
  >
    Search
  </button>
</form>
`;
